src/parser.js

```diff
@@ -0,0 +1,520 @@
+
+const SpecialHeadingDepth = 2
+const SpecialHeadings = /^Arguments|Events|Examples/
+
+const VisibilityRegexStr = '^\\s*([a-zA-Z]+):\\s*'
+const VisibilityRegex = new RegExp(VisibilityRegexStr)
+const ReturnsRegex = new RegExp(`(${VisibilityRegexStr})?\\s*Returns`)
+const ArgumentListItemRegex = /^\s*`([\w\\.-]+)`(\s*[:-])?(\s*\(optional\))?\s*/
+
+/*
+Section: Parsing
+
+Translating things from markdown into our json format.
+*/
+
+// Public: Parses a docString
+//
+// * `docString` a string from the documented object to be parsed
+//
+// Returns a {Doc} object
+const parse = function (docString) {
+  const lexer = new marked.Lexer()
+  const tokens = lexer.lex(docString)
+  const firstToken = _.first(tokens)
+
+  if (!firstToken || (firstToken.type !== 'paragraph')) {
+    throw new Error('Doc string must start with a paragraph!')
+  }
+
+  const doc = new Doc(docString)
+
+  _.extend(doc, parseSummaryAndDescription(tokens))
+
+  while (tokens.length) {
+    let args, events, examples, returnValues, titledArgs
+    if ((titledArgs = parseTitledArgumentsSection(tokens))) {
+      if (doc.titledArguments == null) { doc.titledArguments = [] }
+      doc.titledArguments.push(titledArgs)
+    } else if ((args = parseArgumentsSection(tokens))) {
+      doc.arguments = args
+    } else if ((events = parseEventsSection(tokens))) {
+      doc.events = events
+    } else if ((examples = parseExamplesSection(tokens))) {
+      doc.examples = examples
+    } else if ((returnValues = parseReturnValues(tokens, true))) {
+      doc.setReturnValues(returnValues)
+    } else {
+      // These tokens are basically in no-mans land. We'll add them to the
+      // description so they dont get lost.
+      const extraDescription = generateDescription(tokens, stopOnSectionBoundaries)
+      doc.description += `\n\n${extraDescription}`
+    }
+  }
+
+  return doc
+}
+
+var parseSummaryAndDescription = function (tokens, tokenCallback) {
+  if (tokenCallback == null) { tokenCallback = stopOnSectionBoundaries }
+  let summary = ''
+  let description = ''
+  let visibility = 'Private'
+
+  let rawVisibility = null
+  let rawSummary = tokens[0].text
+  if (rawSummary) {
+    const visibilityMatch = VisibilityRegex.exec(rawSummary)
+    if (visibilityMatch) {
+      visibility = visibilityMatch[1]
+      rawVisibility = visibilityMatch[0]
+      if (rawVisibility) { rawSummary = rawSummary.replace(rawVisibility, '') }
+    }
+  }
+
+  if (isReturnValue(rawSummary)) {
+    const returnValues = parseReturnValues(tokens, false)
+    return {summary, description, visibility, returnValues}
+  } else {
+    summary = rawSummary
+    description = generateDescription(tokens, tokenCallback)
+    if (rawVisibility) { description = description.replace(rawVisibility, '') }
+    return {description, summary, visibility}
+  }
+}
+
+var parseArgumentsSection = function (tokens) {
+  const firstToken = _.first(tokens)
+  if (firstToken && (firstToken.type === 'heading')) {
```

Comment:
No need for extra parentheses

Reply:
Whoops, missed that one. I don't know why `standard` doesn't enable that rule...